File: src/components/JSONHandler/src/Speak.cpp
#include "JSONHandler/Speak.h"
#include "JSONHandler/RPC2Marshaller.h"

using namespace RPC2Communication;

Speak::Speak()
:RPC2Request( RPC2Marshaller::METHOD_SPEAK_REQUEST )
{}
 
Speak::~Speak()
{}

const std::vector<TTSChunk> & Speak::getTTSChunks() const
{
    return mTTSChunks;
}
 
void Speak::setTTSChunks( const std::vector<TTSChunk>& s )
{
    mTTSChunks = s;
}

File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/Speak.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;
import java.util.Vector;

import com.ford.syncV4.proxy.RPCRequest;
import com.ford.syncV4.proxy.constants.Names;

public class Speak extends RPCRequest {

    public Speak() {
        super("Speak");
    }
    public Speak(Hashtable hash) {
        super(hash);
    }
    public Vector<TTSChunk> getTtsChunks() {
    	if (parameters.get(Names.ttsChunks) instanceof Vector<?>) {
	        Vector<?> list = (Vector<?>)parameters.get(Names.ttsChunks);
	        if (list != null && list.size() > 0) {
	            Object obj = list.get(0);
	            if (obj instanceof TTSChunk) {
	                return (Vector<TTSChunk>) list;
	            } else if (obj instanceof Hashtable) {
	                Vector<TTSChunk> newList = new Vector<TTSChunk>();
	                for (Object hashObj : list) {
	                    newList.add(new TTSChunk((Hashtable)hashObj));
	                }
	                return newList;
	            }
	        }
    	}
        return null;
    }
    public void setTtsChunks( Vector<TTSChunk> ttsChunks ) {
        if (ttsChunks != null) {
            parameters.put(Names.ttsChunks, ttsChunks );
        }
    }
}
